### DISCUSSION TITLE
Tcldot message

### DISCUSSION THREAD

**Discussion by 3droot** [B-Lang-org/bdw#7] [Q&A] [answered]
I have installed bdw, when I run it, terminals tells me:
"Tcldot packages are not available on this machine
Information: can't find package Tcldot 2.21"

How can I install tcldot?

**Comment by quark17** [comment, marked as answer]
Tcldot is only needed for a minor feature.  If you're not using that feature (and I don't use it myself), then you can ignore the warning.

If you do want to install it, it is part of Graphviz.  On Ubuntu, there is a `graphviz` package that you can install.  Or you can download from [graphviz.org](http://graphviz.org).  However, the latest version is greater than 2.21, and I don't know if that will be an issue.  The CI is not currently installing `graphviz`, so that feature is not being tested (but we can look into changing that).

**Reply by 3droot** [reply]
I compiled the graphviz, there is one condition that the BDW must recognize the shared object of the tdldot.

**Reply by quark17** [reply]
If the package is not installed in a location that is automatically visible to Tcl, then you will need to add its location to the search path for packages.  This can be done by adding a line like this:
```
lappend auto_path /usr/lib/tcltk/graphviz/tcl
```
to a file called `.bluetclrc` in your home directory.  (You can also put it in a `.tclrc` file.  Or, I believe, you can put it in a file called `bdw_init.tcl` in either your home directory or the directory where BDW is run.  It depends on whether you want the package available when using any Tcl, just Bluetcl, or specifically just BDW.)

This is described in the BDW User Guide, in a section titled "Viewing graphs and installing Tcldot".  There, it explains that you can test your installation, by running `wish` or `tclsh` (or even `bluetcl`) interactively, and giving it the command:
```
% package require Tcldot                                                                                                                     
```
If this fails, you can see what the search path is with this command:
```
% puts $auto_path
```
If graphviz is not found in those directories, then you can add it with `lappend` in a `.bluetclrc` file, and restart BDW.

**Comment by jahagirdar** [comment]
Solution for ubuntu
`apt install libgv-tcl`
add this line to $HOME/.bluetclrc ` lappend auto_path /usr/lib/x86_64-linux-gnu/graphviz/tcl/`
